// @stimulus-components/color-picker/dist/stimulus-color-picker.mjs
import { Controller } from "@hotwired/stimulus";
import Pickr from "@simonwep/pickr";
const _ColorPicker = class _ColorPicker extends Controller {
  initialize() {
    this.onSave = this.onSave.bind(this);
  }
  connect() {
    this.picker = Pickr.create({
      el: this.buttonTarget,
      theme: this.themeValue,
      default: this.inputTarget.value,
      swatches: this.swatches,
      components: this.componentOptions
    }), this.picker.on("save", this.onSave);
  }
  disconnect() {
    this.picker.destroy();
  }
  onSave(color) {
    this.inputTarget.value = null, color && (this.inputTarget.value = color.toHEXA().toString()), this.picker.hide();
  }
  get componentOptions() {
    return {
      preview: !0,
      hue: !0,
      interaction: {
        input: !0,
        clear: !0,
        save: !0
      }
    };
  }
  get swatches() {
    return [
      "#A0AEC0",
      "#F56565",
      "#ED8936",
      "#ECC94B",
      "#48BB78",
      "#38B2AC",
      "#4299E1",
      "#667EEA",
      "#9F7AEA",
      "#ED64A6"
    ];
  }
};
_ColorPicker.targets = ["button", "input"], _ColorPicker.values = {
  theme: {
    type: String,
    default: "classic"
  }
};
let ColorPicker = _ColorPicker;
export {
  ColorPicker as default
};
